addon/utils/debug-logger.js

```diff
@@ -13,25 +13,90 @@
  * Logging preferences are persisted in local storage, and you'll need to reload
  * the page for changes to take effect.
  */
-export default function debugLogger(key) {
-  return key ? window.debug(key) : instanceLogger;
+
+/**
+ * The default logging functions
+ * @type {{debug: *, info: *, warn: *, error: *}}
+ */
+let LOGGERS = {
+  'debug': console.log.bind(console),
+  'info' : console.info.bind(console),
+  'warn' : console.warn.bind(console),
+  'error': console.error.bind(console)
+};
+
+/**
+ * These functions wrap the loggers, so we can perform runtime lookups
+ * of the loggers, and change them on the fly.
+ * @type {{debug: (function(): *), info: (function(): *), warn: (function(): *), error: (function(): *)}}
+ */
+const WRAPPERS = {
+  'debug': (...args) => LOGGERS['debug'](...args),
+  'info' : (...args) => LOGGERS['info'](...args),
+  'warn' : (...args) => LOGGERS['warn'](...args),
+  'error': (...args) => LOGGERS['error'](...args)
+};
+
+/**
+ * Replaces the logging function for the various log types.
+ * Expects an object with keys of: debug, info, warn, or error
+ * The values should be a function.
+ *
+ * @function overrideLoggers
+ * @param {Object} loggers a hash as described above
+ */
+export function overrideLoggers(loggers) {
+  LOGGERS = loggers;
+  for (const type of Object.keys(loggers)) {
```

Comment:
When Babel compiles a `for/of` loop, it assumes the existence of `Symbol.iterator` — this will cause issues for anyone supporting older browsers without an ES6 shim.